pvlib/modelchain.py

```diff
@@ -1090,39 +1244,89 @@ def _verify_df(self, data, required):
         ------
         ValueError if any of required are not in data.columns.
         """
-        if not set(required) <= set(data.columns):
-            raise ValueError(
-                f"Incomplete input data. Data needs to contain {required}. "
-                f"Detected data contains: {list(data.columns)}")
-        return
+        def _verify(data):
+            if not set(required) <= set(data.columns):
+                raise ValueError(
+                    "Incomplete input data. Data needs to contain "
+                    f"{required}. Detected data contains: "
+                    f"{list(data.columns)}")
+        if not isinstance(data, tuple):
+            _verify(data)
+        else:
+            for array_data in data:
+                _verify(array_data)
```

Comment:
It might be useful to include the tuple index in the ValueError. One untested approach.... ``for i, array_data in data: _verify(array_data, i=i)``, with 

```
        def _verify(data, i=None):
            if not set(required) <= set(data.columns):
                tuple_txt = ' ' if i is None else f'in element {i} '
                raise ValueError(
                    "Incomplete input data. Data needs to contain "
                    f"{required}. Detected data {tuple_txt}contains: "
                    f"{list(data.columns)}")
```